tests/test_gene_association_file.py

```diff
@@ -0,0 +1,151 @@
+    return pd.DataFrame([
+        {"Evidence_Code": "ISS", "publications": "GO_REF:0000024"},
+        {"Evidence_Code": "ND", "publications": "PMID:UNKNOWN"},
+    ])
+
+
+@pytest.mark.unit
+def test_go_annotation_processed(tmp_path):
+    """Test that the processed GO annotation CSV file has correct columns."""
+    output_file = tmp_path / "new_annotations1.csv"
+    process_go_annotations("annotations_data100.csv", str(output_file))
+
+    assert output_file.exists(), "Output file not created."
+    df = pd.read_csv(output_file)
+
+    expected_columns = [
+        "subject", "predicate", "object", "publications", "Evidence_Code", "supporting_objects",
+        "annotation_date", "primary_knowledge_source", "aggregator", "protocol_id", "negated", "evidence_type"
+    ]
+
+    for col in expected_columns:
+        assert col in df.columns, f"Column {col} missing in output."
+
+
+@pytest.mark.integration
+def test_io_behavior(temp_csv_files):
+    """Integration test for I/O behavior of process_go_annotations."""
+    input_path, output_path = temp_csv_files
+
+    process_go_annotations(str(input_path), str(output_path))
+
+    assert output_path.exists(), "Output file not created."
+    df_out = pd.read_csv(output_path)
+
+    assert not df_out.empty, "Output DataFrame is empty."
+
+    expected_columns = [
+        "subject", "predicate", "object", "publications", "Evidence_Code",
+        "supporting_objects", "annotation_date", "primary_knowledge_source",
+        "aggregator", "protocol_id", "negated", "evidence_type"
+    ]
+
+    for col in expected_columns:
+        assert col in df_out.columns, f"Column {col} missing in output."
+
+
+@pytest.mark.unit
+def test_column_types(temp_csv_files):
+    """Test the column data types of the processed annotations."""
+    input_path, output_path = temp_csv_files
+
+    process_go_annotations(str(input_path), str(output_path))
+    df = pd.read_csv(output_path)
+
+    expected_types = {
+        "subject": str,
+        "predicate": str,
+        "object": str,
+        "publications": str,
+        "supporting_objects": str,
+        "annotation_date": str,
+        "primary_knowledge_source": str,
+        "aggregator": str,
+        "protocol_id": str,
+        "evidence_type": str,
+        "negated": bool
+    }
+
+    for column, expected_type in expected_types.items():
+        invalid = df[column].dropna().apply(lambda x: not isinstance(x, expected_type))
+        assert not invalid.any(), f"Column {column} has invalid data types."
+
+
+@pytest.mark.unit
+def test_negated_logic(temp_csv_files):
+    """Test negation logic of predicate column."""
+    input_path, output_path = temp_csv_files
+
+    process_go_annotations(str(input_path), str(output_path))
+    df = pd.read_csv(output_path)
+
+    assert df.loc[0, "negated"] == False, "Negation logic failed for allowed predicate."
+
+
+@pytest.mark.unit
+def merge_evidence_mapping(df, df2):
```

Comment:
Rather than recapitulate logic from your original script, it's better to separate out the logic that you want to test in the script into its own function so that you can test it directly.